src/Signal.lua

```diff
@@ -47,15 +48,46 @@ function Signal.new()
 end
 
 function Signal:connect(callback)
+	if typeof(callback) ~= "function" then
+		error("Expected the listener to be a function.")
+	end
+
+	-- TODO: subscribe can't happen while store isDispatching, but can't get that store state here
+	if self._store and self._store.isDispatching then
+		error(
+			'You may not call store.changed:connect() while the reducer is executing. ' ..
+			  'If you would like to be notified after the store has been updated, subscribe from a ' ..
+			  'component and invoke store:getState() in the callback to access the latest state. '
```

Comment:
Are we taking into account the realities of how `rodux` is used with this messaging? Most of Roblox's internal use of `rodux` is through [`roact-rodux`](https://github.com/roblox/roact-rodux), so we ought to make sure that whatever error gets spit out from all of this would be understandable coming through that interface.

We might also consider looking there for more opportunities to add validation or error handling.

Reply:
Good call on fortifying roact-rodux as well. This messaging comes from upstream redux, so it should be web search-able.

Reply:
Researched roact-rodux, and there's not much more fortification to do there that wouldn't duplicate what's in here.